basictracer/tracer.py

```diff
@@ -11,7 +12,7 @@
 
 class BasicTracer(Tracer):
 
-    def __init__(self, recorder=None, sampler=None):
+    def __init__(self, recorder=None, sampler=None, scope_manager=None):
```

Comment:
It's reasonable defining a `scope_manager` in the constructor, since a `Tracer` should have one and only one `ScopeManager` for the rest of the execution. Switching it at runtime could lead to unexpected behaviors.

Reply:
I agree here with you. My only question is whether we should provide, on default, a noop `ScopeManager` by default or not...